lib/registry.js

```diff
@@ -236,7 +236,7 @@ Registry.prototype.configureModel = function(ModelCtor, config) {
     ModelCtor.attachTo(config.dataSource);
     debug('Attached model `%s` to dataSource `%s`',
       modelName, config.dataSource.name);
-  } else if (config.dataSource === null) {
+  } else if (config.dataSource === null || config.dataSource === false) {
```

Comment:
It's intentional as `undefined` means the property is not set in json and we want to warn the user.